### ISSUE TITLE
proxy-postsubmit failed to build proxy

### ISSUE THREAD

**Issue by yangminzhu** [istio/istio#15714]
It seems the postsubmit failed since https://github.com/istio/proxy/pull/2310 with the following message:
```
Target //src/envoy:envoy_tar up-to-date:
  /home/bootstrap/.cache/bazel/_bazel_bootstrap/45ae47b4a0e12d1e81c831ece04d820d/execroot/__main__/bazel-out/k8-opt/bin/src/envoy/envoy_tar.tar.gz
INFO: Elapsed time: 939.158s, Critical Path: 213.68s
INFO: 2258 processes: 2258 processwrapper-sandbox.
INFO: Build completed successfully, 3056 total actions
INFO: Build completed successfully, 3056 total actions
+ BAZEL_TARGET=bazel-bin/src/envoy/envoy_tar.tar.gz
+ cp -f bazel-bin/src/envoy/envoy_tar.tar.gz envoy-symbol-00bd27fad4e78cf5d803f9b39010b71103eda183.tar.gz
cp: cannot stat 'bazel-bin/src/envoy/envoy_tar.tar.gz': No such file or directory
```

@utka Could you help take a look? This is blocking Istio from updating the proxy sha which is needed for https://github.com/istio/istio/issues/15631 for 1.3, thank you!

**Comment by utka** [comment]
I removed privileged from container and bazel was not able to create symlink. Adding it now.

**Comment by yangminzhu** [comment]
@utka It seems the postsubmit is still failing at https://github.com/istio/proxy/pull/2324, is it because the postsubmit just doesn't pick the latest change? Do we need to send out another PR to trigger this? Thank you.

cc @crazyxy

**Comment by utka** [comment]
No, it picked up the change, but it is still permission denied error, but now with creating file itself.

**Comment by utka** [comment]
I think there is still wrong permissions on prow cluster 
When I run gcloud container clusters describe  prow --zone=us-west1-a | grep read
It returns:
 - https://www.googleapis.com/auth/devstorage.read_only
    - https://www.googleapis.com/auth/devstorage.read_only
    - https://www.googleapis.com/auth/devstorage.read_only
    - https://www.googleapis.com/auth/devstorage.read_only
    - https://www.googleapis.com/auth/service.management.readonly
I think it should be read_write instead.

**Comment by myidpt** [comment]
@cjwagner Hi Cole, could you help to look into this issue? Thanks!

**Comment by cjwagner** [comment]
It is unclear to me what setting change is being requested or what the actual error is/where it is coming from. If this is a permissions error, why did it start happening after https://github.com/istio/proxy/pull/2310?

**Comment by utka** [comment]
I don't think it is related, the failure which is happening right now is insufficient permissions to copy data to bucket. I had the same problem with another project some time ago and it turned out to be problem with scope (read_only vs read_write) that's why I think that's the same issue. 
I already submitted PR which returns privileged to the container and it did not help https://github.com/istio/test-infra/pull/1529/files

**Comment by cjwagner** [comment]
Ah, it sounds like the job needs to mount and use a GCP service account token with permission to upload to the bucket.
Previously the boostrap.py script did this before running the job to upload metadata and logs to GCS and the service account happened to be available to the test, but now the pod utilities do this in a separate container to isolate the service account from the test container.
If the job just needs to upload some artifacts to its job directory it can put files in the `$ARTIFACTS` directory and they will be automatically uploaded. If the job instead needs to upload artifacts to some other specific location then it should be providing a service account with the appropriate credentials and using that.

**Comment by myidpt** [comment]
Thanks a lot folks. @cjwagner can I assume that you will help resolve this issue? :)

**Comment by cjwagner** [comment]
> Thanks a lot folks. @cjwagner can I assume that you will help resolve this issue? :)

It is being addressed here: https://github.com/istio/test-infra/pull/1579

**Comment by yangminzhu** [comment]
It seems the post submit is still failing, https://storage.googleapis.com/istio-prow/logs/proxy-postsubmit/642/build-log.txt

```
INFO: Build completed successfully, 2317 total actions
+ [[ -n gs://istio-artifacts/proxy/25c81550a1091f8224ece98876d558ba6769a2b4/artifacts/debs ]]
+ gsutil -m cp -r /home/prow/go/src/istio.io/proxy/bazel-bin/tools/deb/istio-proxy.deb gs://istio-artifacts/proxy/25c81550a1091f8224ece98876d558ba6769a2b4/artifacts/debs/
CommandException: No URLs matched: /home/prow/go/src/istio.io/proxy/bazel-bin/tools/deb/istio-proxy.deb
CommandException: 1 file/object could not be transferred.
Makefile:72: recipe for target 'artifacts' failed
make: *** [artifacts] Error 1
```

**Comment by yangminzhu** [comment]
@utka @cjwagner Could you take a look? thank you.

**Comment by kyessenov** [comment]
The latest finding is that `bazel` fails to create the convenience symbolic links (e.g. `bazel-bin`). Did we change permissions in the container?

**Comment by myidpt** [comment]
Raise this issue to P0 since we have P0 task blocked by this.

**Comment by PiotrSikora** [comment]
`proxy-postsubmit` should be fixed by istio/proxy#2345, but it would be a good idea to fix the underlying issue with symlinks. How come we're mixing `/home/bootstrap` and `/home/prow`?

**Comment by duderino** [comment]
Nice one Piotr!

**Comment by yangminzhu** [comment]
@PiotrSikora Thanks for helping on this. It seems the post-submit works after istio/proxy#2343 but it failed again on https://github.com/istio/proxy/pull/2346, would you mind to take a second look?
```
ERROR: /home/bootstrap/.cache/bazel/_bazel_bootstrap/45ae47b4a0e12d1e81c831ece04d820d/external/envoy/source/extensions/common/wasm/BUILD:36:1: C++ compilation of rule '@envoy//source/extensions/common/wasm:wasm_lib' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffunction-sections ... (remaining 276 argument(s) skipped)
 Use --sandbox_debug to see verbose messages from the sandbox
external/envoy/source/extensions/common/wasm/wasm.cc: In function 'std::shared_ptr<Envoy::Extensions::Common::Wasm::Wasm> Envoy::Extensions::Common::Wasm::createWasmInternal(absl::string_view, const envoy::config::wasm::v2::VmConfig&, absl::string_view, Envoy::Upstream::ClusterManager&, Envoy::Event::Dispatcher&, Envoy::Api::Api&, Envoy::Stats::Scope&, const Envoy::LocalInfo::LocalInfo&, const envoy::api::v2::core::Metadata*, Envoy::Stats::ScopeSharedPtr, std::unique_ptr<Envoy::Extensions::Common::Wasm::Context>)':
external/envoy/source/extensions/common/wasm/wasm.cc:2391:12: error: variable 'context' set but not used [-Werror=unused-but-set-variable]
   Context* context;
            ^~~~~~~
cc1plus: all warnings being treated as errors
Target //tools/deb:istio-proxy failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 593.113s, Critical Path: 98.18s
INFO: 1881 processes: 1881 linux-sandbox.
FAILED: Build did NOT complete successfully
FAILED: Build did NOT complete successfully
Makefile:72: recipe for target 'artifacts' failed
make: *** [artifacts] Error 1
```

**Comment by PiotrSikora** [comment]
Sigh, it looks that as part of `proxy-postsubmit` we have:
- `scripts/release-binary.sh` step that builds binaries and Debian packages and pushes them to `gs://istio-build/proxy`,
- `make artifacts` step that builds Debian package (using slightly different options) and pushes it to `gs://istio-artifacts/proxy/${GIT_SHA}/artifacts/debs`.

But the real issue is that we use `clang` everywhere, but `make artifacts` is being built using `gcc`. I'll fix this in a second.

**Comment by PiotrSikora** [comment]
Fixed in istio/proxy#2347.
